src/invariant/test/ConservationOfLumensTests.cpp

```diff
@@ -202,6 +217,17 @@ TEST_CASE("Account balances unchanged without inflation",
         }
 
         auto keepEnd = entries2.begin() + N / 2;
+
+        // We're going to increase LedgeHeader::totalCoins by the total balance
+        // of the entries starting at `keepEnd` when we call `updateBalances()`
+        // below, so check whether that would overflow, and if so, bump
+        // `keepEnd` until it wouldn't overflow.
+        while (getTotalBalance(std::vector<LedgerEntry>(
```

Comment:
I think this works, but I don't think it fundamentally solves the problem. The root cause of this bug is that we are misusing `getTotalBalance`.

What the test is trying to do is delete approximately half the entries, transferring the balance randomly to other entries that are not being deleted. This doesn't actually change `totalCoins`. But `getTotalBalance` does change `totalCoins`.

Reply:
> The root cause of this bug is that we are misusing getTotalBalance.
>
> What the test is trying to do is delete approximately half the entries, transferring the balance randomly to other entries that are not being deleted. This doesn't actually change totalCoins. But getTotalBalance does change totalCoins.

I think you mean `updateBalances`, not `getTotalBalance`.

Maybe we should just give the three-argument version of `updateBalances()` a fourth, `bool`, argument telling it whether to update `totalCoins` or not?  The other callers _do_ mean to be updating `totalCoins`.  Crude, maybe, but simple.

Reply:
> Maybe we should just give the three-argument version of updateBalances() a fourth, bool, argument telling it whether to update totalCoins or not? The other callers do mean to be updating totalCoins. Crude, maybe, but simple.

I think this works and it doesn't look completely appalling, so I've tried it.

Reply:
Yes, I definitely meant `updateBalances`!